File: bot-shared-library/src/main/java/gg/mineral/bot/impl/thread/MineralThreadFactory.java
package gg.mineral.bot.impl.thread;

import lombok.val;
import org.eclipse.jdt.annotation.NonNull;

import java.util.concurrent.ThreadFactory;

public class MineralThreadFactory implements ThreadFactory {

    @Override
    public Thread newThread(@NonNull Runnable r) {
        val t = new Thread(r);
        t.setName("MineralThread-" + t.getId());
        t.setUncaughtExceptionHandler((thread, throwable) -> {
            System.err.println("Uncaught exception in thread " + thread.getName());
            throwable.printStackTrace();
        });
        return t;
    }

}

File: bot-shared-library/src/main/kotlin/gg/mineral/bot/impl/thread/MineralThreadFactory.kt
package gg.mineral.bot.impl.thread

import java.util.concurrent.ThreadFactory

class MineralThreadFactory(val suffix: String) : ThreadFactory {
    override fun newThread(r: Runnable): Thread {
        val t = Thread(r)
        t.name = "MineralThread-" + t.id + "-" + suffix
        t.uncaughtExceptionHandler =
            Thread.UncaughtExceptionHandler { thread: Thread, throwable: Throwable ->
                System.err.println("Uncaught exception in thread " + thread.name)
                throwable.printStackTrace()
            }
        return t
    }
}
